src/components/Dropdown/Dropdown.tsx

```diff
@@ -411,9 +412,13 @@ const Dropdown = ({
             break;
           case 'ArrowUp':
             if (focusedElement && focusedElement.id.match(`${name}-option-.*`)) {
-              const sibling = focusedElement.previousElementSibling as HTMLElement | null;
-              if (sibling) {
-                sibling.focus();
+              const row = focusedElement.parentNode as HTMLElement | undefined;
```

Comment:
I hadn't even thought of the implications with keyboard controls focusing options that aren't being rendered - nicely done!